cmd/finch/nerdctl.go

```diff
@@ -122,6 +167,93 @@ func (nc *nerdctlCommand) shouldReplaceForHelp(cmdName string, args []string) bo
 	return false
 }
 
+func argIsEnv(arg string) bool {
+	if strings.HasPrefix(arg, "-e") || (strings.HasPrefix(arg, "--env") && !strings.HasPrefix(arg, "--env-file")) {
+		return true
+	}
+	return false
+}
+
+func handleEnv(arg, arg2 string) (bool, []string) {
+	stdlib := system.NewStdLib()
+	var (
+		envs   []string
+		envVar string
+		skip   bool
+	)
+	switch arg {
+	case "-e", "--env":
+		skip = true
+		envVar = arg2
+	default:
+		// flag and value are in the same string
+		if strings.HasPrefix(arg, "-e") {
+			envVar = arg[2:]
+		} else {
+			// only other case is "--env="; skip that prefix
+			envVar = arg[6:]
+		}
+	}
+
+	if !strings.Contains(envVar, "=") {
+		// need to lookup value from OS environment, and only set if it exists in the env
+		if val, ok := stdlib.LookupEnv(envVar); ok {
+			envs = append(envs, fmt.Sprintf("%s=%s", envVar, val))
+		}
+	} else {
+		envs = append(envs, envVar)
+	}
+	return skip, envs
+}
+
+func handleEnvFile(arg, arg2 string) (bool, []string, error) {
+	var (
+		filename string
+		envs     []string
+		skip     bool
+	)
+
+	switch arg {
+	case "--env-file":
+		skip = true
+		filename = arg2
+	default:
+		filename = arg[11:]
+	}
+
+	file, err := os.Open(filepath.Clean(filename))
```

Comment:
personally I don't see how this adds any value at this point as test coverage is solid and there is no added value to this capability working or not working with more mocking around the handling of a simple file. If the maintainers feel strongly about it maybe it can be done in a follow-on improvement?

Reply:
> how this adds any value at this point

Mainly mitigate https://github.com/runfinch/finch/pull/158#issuecomment-1413129951 and make the file-handling code consistent.

> maybe it can be done in a follow-on improvement?

Sure, I can do a follow-up PR.